internal/controller/dnsrecord_controller.go

```diff
@@ -391,9 +381,9 @@ func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, curren
 		// generation has not changed but there are changes.
 		// implies that they were overridden - bump write counter
 		requeueTime = randomizedValidationRequeue
+		// BUG: https://github.com/Kuadrant/dns-operator/issues/664 The premature checks stops this from running.
```

Comment:
I don't think it is a bug. 
If we are here, it means we did write to the provider on the same generation. Premature logic, indeed, prevents us from getting here, but we need this specifically for the moments when premature logic allows us to do a full loop.

Reply:
The problem is the premature logic stops this reconciling if the status has being updated. When the generation changes the metric gets reset back to zero. 

That issue is also marked as a bug.